src/main/java/com/openlattice/chronicle/services/ChronicleServiceImpl.java

```diff
@@ -553,4 +551,81 @@ public void refreshStudyInformation() {
             return ImmutableSetMultimap.of();
         }
     }
+
+    @Override
+    public UUID getParticipantEntityKeyId( String participantId, UUID participantsEntitySetId, SearchApi searchApi ) {
+        DataSearchResult result = searchApi.executeEntitySetDataQuery(
+                participantsEntitySetId,
+                new SearchTerm( participantIdPropertyTypeId.toString() + ":\"" + participantId + "\"", 0, 1 )
+        );
+        if ( result.getHits().size() != 1 ) {
+            return null;
+        }
+        SetMultimap<FullQualifiedName, Object> data = result.getHits().iterator().next();
+        if ( data.containsKey( INTERNAL_ID_FQN ) ) {
+            return UUID.fromString( data.get( INTERNAL_ID_FQN ).iterator().next().toString() );
+        }
+        return null;
+    }
+
+    @Override
+    public ParticipationStatus getParticipationStatus( UUID studyId, String participantId ) {
+
+        EdmApi edmApi;
+        SearchApi searchApi;
+
+        try {
+            ApiClient apiClient = apiClientCache.get( ApiClient.class );
+            edmApi = apiClient.getEdmApi();
+            searchApi = apiClient.getSearchApi();
+        } catch ( ExecutionException e ) {
+            logger.error( "Unable to load apis." );
+            return ParticipationStatus.NOT_ENROLLED;
+        }
+
+        String participantsEntitySetName = ChronicleServerUtil.getParticipantEntitySetName( studyId );
+        UUID participantsEntitySetId = edmApi.getEntitySetId( participantsEntitySetName );
+        if ( participantsEntitySetId == null ) {
+            logger.error( "unable to get the participants EntitySet id for studyId = {}", studyId );
+            return ParticipationStatus.NOT_ENROLLED;
+        }
+
+        UUID participantEntityKeyId = getParticipantEntityKeyId( participantId, participantsEntitySetId, searchApi );
+        if ( participantEntityKeyId == null ) {
+            logger.error( "unable to load participant entity key id for participantId = {}", participantId );
+            return ParticipationStatus.NOT_ENROLLED;
+        }
+
+        Map<UUID, List<NeighborEntityDetails>> neighborResults = searchApi.executeFilteredEntityNeighborSearch(
+                participantsEntitySetId,
+                new EntityNeighborsFilter(
+                        Sets.newHashSet( participantEntityKeyId ),
+                        java.util.Optional.of( Sets.newHashSet( participantsEntitySetId ) ),
+                        java.util.Optional.of( Sets.newHashSet( studyEntitySetId ) ),
```

Comment:
slightly less overhead to do `ImmutableSet.of()` / `ImmutableSet.of( studyEntitySetId )` / `ImmutableSet.of( participatedInEntitySetId )`